### PR TITLE
Collection form configures review workflow

### PR DESCRIPTION
**Pull request by jmartin-sul**
## Why was this change made?

Allow configuration of the review workflow on collections.

connects to #348

This PR doesn't quite fully implement what's shown in the mockup and described in the design doc.  In particular:
* the enable/disable button is just a basic radio button instead of the cool switch in the mockup (because i wasn't sure how to do that, and wanted to get everything else working).
* the reviewers list field isn't hidden based on whether the radio button is enabled (i wasn't sure how to implement that idiomatically in this code base).

I'm happy to circle back on those two shortfalls in this PR, or in a follow-on PR, whichever seems most appropriate.  Happy to write the ticket for the next part, if doing that in a separate PR is preferable.

## How was this change tested?

* new unit tests
* manual testing locally

## Which documentation and/or configurations were updated?

n/a

### CODE REVIEW

**Comment by jmartin-sul** [comment]
re: the things that need to be revisited, per the PR description:

#### mockup
Screen Shot 2020-11-09 at 11 54 55 AM


#### current implementation
Screen Shot 2020-11-09 at 11 55 20 AM

#### from the design doc
> Table 6: Review Workflow fa-pencil [&#xf040;]
> * Status, with options of "On" or "Off"
> * Reviewers
>   * Only display reviewers if the workflow is on

File: app/forms/collection_form.rb [hunk 1]
```diff
@@ -3,34 +3,51 @@
...
 class CollectionForm < Reform::Form
+  extend T::Sig
+
   property :name
   property :description
   property :contact_email
-  property :managers, prepopulator: lambda { |_options|
-    self.managers = model.creator.sunetid
-  }
+  property :managers, default: -> { model.creator.sunetid }
```
Comment (jmartin-sul) [inline app/forms/collection_form.rb:11]: i stumbled on what seemed like the need for this change when i copypasta'ed the `prepopulator` pattern  for the reviewers field, and noticed that it was only ever presenting the default value, even if i'd saved an updated value to the DB.

File: app/forms/collection_form.rb [hunk 2]
```diff
@@ -3,34 +3,51 @@
...
   property :access, default: 'world'
   property :creator, writable: false
   property :depositor_sunets, virtual: true, prepopulator: lambda { |_options|
     self.depositor_sunets = depositor_sunets_from_model.join(', ')
   }
+  # TODO: review_enabled==false just nils out reviewers on save? review_enabled
+  # starts true if reviewers not nil, false otherwise?
```
Comment (jmartin-sul) [inline app/forms/collection_form.rb:18]: d'oh, cruft, removing...

File: app/components/collections/form_component.html.erb [hunk 3]
```diff
@@ -75,5 +75,37 @@
...
+  <section>
+    <header>Review Workflow</header>
+
+    <div class="mb-3 row">
+      <%= form.label :review_enabled, class: 'col-sm-9 col-form-label' do %>
+        A reviewer checks a deposit for clarity, accuracy, and completeness before approving the publication. Only one Reviewer
+        is required to review each deposit. A Reviewer can edit a deposit or return it to the Depositor with a request for changes.
+        The Depositor will receive a notification when their deposit is returned or approved.
+      <% end %>
+      <div class="col-sm-3">
```
Comment (mjgiarlo) [inline app/components/collections/form_component.html.erb:87]: I'm not saying you need to change this PR, but this is how you can render a switch in Bootstrap: https://getbootstrap.com/docs/4.2/components/forms/#switches

File: app/components/collections/form_component.html.erb [hunk 4]
```diff
@@ -75,5 +75,37 @@
...
+        </div>
+      </div>
+    </div>
+
+    <!-- TODO: only display sunet entry field if reviewer workflow is enabled -->
+    <p>Enter SUNet IDs of participants. Separate each one with a comma and a space. (e.g. janelath, lelandst)</p>
+    <div class="mb-3 row">
+      <%= form.label :reviewer_sunets, 'Additional Reviewers', class: 'col-sm-2 col-form-label' %>
+      <div class="col-sm-10">
+        <%= form.text_field :reviewer_sunets, class: "form-control", required: false %> <!-- TODO: should prob have at least one required if enabled? -->
```
Comment (mjgiarlo) [inline app/components/collections/form_component.html.erb:104]: Yeah, probably. I guess there are two strategies here: pure JavaScript validation or use remote form validation (how we're dealing with keywords).

**Review by mjgiarlo — APPROVED** [review]
LGTM. Would like @jcoyne to review before we merge it, given his greater experience with reform forms.

File: app/components/collections/form_component.html.erb (hunk 3 above)
Comment (jcoyne) [reply app/components/collections/form_component.html.erb:87]: Let's make this a new ticket.

File: app/components/collections/form_component.html.erb [hunk 5]
```diff
@@ -75,5 +75,37 @@
...
+          <%= form.label :collection_review_enabled_false, 'Disable Review Workflow', class: "form-check-label" %>
+        </div>
+      </div>
+    </div>
+
+    <!-- TODO: only display sunet entry field if reviewer workflow is enabled -->
+    <p>Enter SUNet IDs of participants. Separate each one with a comma and a space. (e.g. janelath, lelandst)</p>
+    <div class="mb-3 row">
+      <%= form.label :reviewer_sunets, 'Additional Reviewers', class: 'col-sm-2 col-form-label' %>
+      <div class="col-sm-10">
```
Comment (mjgiarlo) [inline app/components/collections/form_component.html.erb:103]: I wonder if this should be `col-sm-7` to make the field's right edge align with the text above it, looking at the mock-up. (7 may be off by one...) Worth a quick try?

**Review by jcoyne — APPROVED** [review]

File: app/components/collections/form_component.html.erb (hunk 3 above)
Comment (mjgiarlo) [reply app/components/collections/form_component.html.erb:87]: @jcoyne OK. You want it or shall I?

Comment (mjgiarlo) [reply app/components/collections/form_component.html.erb:87]: Done: https://github.com/sul-dlss/happy-heron/issues/399

Comment (jmartin-sul) [reply app/components/collections/form_component.html.erb:87]: thanks!  working on it in a follow on PR now...

File: app/components/collections/form_component.html.erb (hunk 5 above)
Comment (jmartin-sul) [reply app/components/collections/form_component.html.erb:103]: will give it a shot...

Comment (mjgiarlo) [reply app/components/collections/form_component.html.erb:103]: Already got it. Thanks!

**Comment by jmartin-sul** [comment]
follow on work written up here:  https://github.com/sul-dlss/happy-heron/issues/402

File: app/components/collections/form_component.html.erb (hunk 3 above)
Comment (jmartin-sul) [reply app/components/collections/form_component.html.erb:87]: looks like i should refresh the page proactively before replying 😄

**Pull request merged**